__tests__/app.test.js

```diff
@@ -112,6 +112,52 @@ describe("GET: /api/articles", () => {
         });
       });
   });
+  
+});
+
+describe("GET: /api/articles/:article_id/comments", () => {
+  test("200: responds with a 200 status and returns all the comments for the specified article_id and orders with by date, ,ost recent first", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("200: has a property of comment count that counts all the comments related to that article id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toHaveLength(11);
+        body.comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id");
+          expect(comment).toHaveProperty("votes");
+          expect(comment).toHaveProperty("created_at");
+          expect(comment).toHaveProperty("author");
+          expect(comment).toHaveProperty("body");
+          expect(comment).toHaveProperty("article_id");
+        });
+      });
+  });
+  test("400: returns a 400 and bad request when passed the wrong data type", () => {
+    return request(app)
+      .get("/api/articles/orange/comments")
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toEqual("Bad Request");
+      });
+  });
+  test("404: returns a 404 and not found when there isnt an article id that matches", () => {
+    return request(app)
+      .get("/api/articles/2000/comments")
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toEqual("Not found");
+      });
+  });
```

Comment:
We're missing one important test here - if a valid article has no comments attached to it. This will also return an empty array the same way an article with a valid id not found would, so we need to add some logic to distinguish the two of them in the model for this endpoint